vmm/src/serial_manager.rs

```diff
@@ -331,19 +331,20 @@ impl SerialManager {
                                     let writer =
                                         unix_stream.try_clone().map_err(Error::CloneUnixStream)?;
                                     reader = Some(
-                                        unix_stream.try_clone().map_err(Error::CloneUnixStream)?,
+                                        unix_stream
                                     );
-
-                                    epoll::ctl(
+                                    if let Some(ref stream) = reader {
```

Comment:
Please rearrange this so that `reader = (Some(unix_stream)` if after the `epoll::ctl` - you can then remove the if block.

Reply:
fixed.